### PR TITLE
Improve contribution guidelines

### PR DESCRIPTION
**Pull request by rwx788**
See [poo#28654](https://progress.opensuse.org/issues/28654).

See also https://github.com/os-autoinst/os-autoinst-needles-opensuse/pull/377

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+Fork the repository and make some changes.
+Once you're done with your changes send a pull request. You have to agree to
+the license. Thanks!
+If you have questions, visit us on irc.freenode.net in #opensuse-factory or
+ask on our mailing list opensuse-factory@opensuse.org
+
+### How to get started
+
+It can be beneficial to learn how openQA job is executed. To setup your own
```
Comment (okurz) [inline CONTRIBUTING.md]: Let's sprinkle in some articles, shall we? :) "learn how *an* openQA job"

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+openQA instance, please refer to the documentation of [openQA project](https://github.com/os-autoinst/openQA).
```
Comment (okurz) [inline CONTRIBUTING.md]: of *the* openQA project

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+Tests developers guide can be found [here](https://github.com/os-autoinst/openQA/blob/master/docs/WritingTests.asciidoc).
```
Comment (okurz) [inline CONTRIBUTING.md]: *A* test developers guide.

I would be very careful with deep links. I guess L12 referencing the documentation is good enough. Or we mention http://open.qa

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+If you are looking for a task to start with, check out [openQA Tests](https://progress.opensuse.org/projects/openqatests/issues/)
```
Comment (okurz) [inline CONTRIBUTING.md]: check out *the* openQA Tests

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+Redmine project. Look for tickets with [easy] or [easy-hack] tags.
+
+### Reporting an issue
+
+As mentioned above, we use Redmine as issue tracking tool. In case you found some
+issue with the tests, please do not hesitate to report them [here](https://progress.opensuse.org/projects/openqatests/issues/).
```
Comment (okurz) [inline CONTRIBUTING.md]: Better replace the `to report them [here](https://progress.opensuse.org/projects/openqatests/issues/)` with `report [a new issue](https://progress.opensuse.org/projects/openqatests/issues/new)`

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+Please, provide issue description and expected results, e.g. last successful
+run on openqa.opensuse.org.
```
Comment (okurz) [inline CONTRIBUTING.md]: Better replace this whole sentence with just a reference to https://progress.opensuse.org/projects/openqav3/wiki/#Defects

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+
+
+If in doubt, use irc channel or mailing list, we will be happy to support you.
```
Comment (okurz) [inline CONTRIBUTING.md]: s/use irc channel or mailing list/contact us/. The contact details had been already provide on top so this should not be repeated

File: CONTRIBUTING.md (hunk 3 above)
Comment (okurz) [reply CONTRIBUTING.md]: what about this comment?

Comment (rwx788) [reply CONTRIBUTING.md]: Ok, now I've changed the line completely.

**Comment by okurz** [comment]
@coolo @foursixnine WDYT in relation to the openQA and os-autoinst project?

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+* Take
+  [example boot.pm](https://github.com/os-autoinst/os-autoinst-distri-example/blob/master/tests/boot.pm)
+  as a template for new files
+* The test code should use simple perl statements, not overly hacky
+  approaches, to encourage contributions by newcomers and test writers which
+  are not programmers or perl experts
+* Update the copyright information with the current year and *SUSE LLC* as the
+  legal entity. For new files make sure to only state the year during which
+  the code was written.
+* Use `my ($self) = @_;` for parameter parsing in methods when accessing the
```
Comment (coolo) [inline CONTRIBUTING.md]: for openQA and os-autoinst I no longer enforce this (and sometimes even actively use my $self = shift; if there is only one argument). But for test code simple rules are better - as most contributors don't necessarly understand perl's argument passing.

Comment (okurz) [reply CONTRIBUTING.md]: Sure. This is not a new rule. This PR is basically about splitting out a file CONTRIBUTING.md which github also offers to read to new contributors. Also it extends the content a bit.

**Comment by rwx788** [comment]
@okurz so are you fine to merge?

**Comment by okurz** [comment]
As there is no further feedback from others let's keep as is

**Pull request merged**